src/storage/sql_table.cpp

```diff
@@ -86,6 +66,57 @@ ProjectionMap SqlTable::ProjectionMapForOids(const std::vector<catalog::col_oid_
   return projection_map;
 }
 
+void SqlTable::Reset() { table_.data_table_->Reset(); }
+
+void SqlTable::CopyTable(const common::ManagedPointer<transaction::TransactionContext> txn,
+                         const common::ManagedPointer<SqlTable> src) {
+  //  uint32_t filled = 0;
+  auto it = src->begin();
+  std::vector<catalog::col_oid_t> col_oids;
+  for (auto &cols : table_.column_map_) {
+    col_oids.push_back(cols.first);
+  }
+  auto pr_init = InitializerForProjectedRow(col_oids);
+  void *buffer = alloca(pr_init.ProjectedRowSize());
```

Comment:
Do we use this style of allocation anywhere else in the system?

Reply:
We don't really use `alloca` in other places, but this is a pointer with a very short lifecycle that we would free almost immediately if we were to put it on the heap so we're just `alloca`ing it instead